flytekit/core/utils.py

```diff
@@ -156,12 +162,22 @@ def _serialize_pod_spec(pod_template: "PodTemplate", primary_container: "task_mo
         # insert a placeholder primary container if it is not defined in the pod spec.
         containers.append(V1Container(name=cast(PodTemplate, pod_template).primary_container_name))
     final_containers = []
+
     for container in containers:
         # In the case of the primary container, we overwrite specific container attributes
         # with the values given to ContainerTask.
         # The attributes include: image, command, args, resource, and env (env is unioned)
+
+        # resolve the image name if it is image spec or placeholder
+        resolved_image = get_registerable_container_image(container.image, settings.image_config)
```

Comment:
then we can use 
```suggestion
        resolved_image = self.get_container(settings).image
```

Reply:
utils.py cannot access `self.get_container`, which is defined in ContainerTask/AutoContainerTask

Reply:
also using `get_container` just to get the image seems to be a bit overkilled because it also includes others like `args`, `cmd`, etc

Reply:
sorry, it's `primary_container.image`

Reply:
`primary_container.image` will become a resolved_image

Reply:
ahh, should we raise an exception here instead? isn't weird to use default image for sidecar?